api/envoy/config/tap/v3/common.proto

```diff
@@ -256,6 +268,21 @@ message StreamingAdminSink {
       "envoy.service.tap.v2alpha.StreamingAdminSink";
 }
 
+// BufferedAdminSink configures a tap output to collect traces without returning them until
+// one of multiple criteria are satisfied.
+// Similar to StreamingAdminSink, it is only allowed to specify the buffered admin output
+// sink if the tap is being configured from the `/tap` admin endpoint.
+message BufferedAdminSink {
+  // Stop collecting traces when the specified number are collected.
+  // If other criteria for ending collection are reached first, this value will not be used.
+  uint64 max_traces = 1 [(validate.rules).uint64 = {gt: 0}];
+
+  // Acts as a fallback to prevent the client from waiting for long periods of time.
+  // After timeout has occurred, a buffer flush will be triggered, returning the traces buffered so far.
```

Comment:
Can you clarify what happens if there are no traces yet? I assume the endpoint will return with some error code or some message?